libs/template/materialize.go

```diff
@@ -48,13 +67,54 @@ func Materialize(ctx context.Context, configFilePath, templateRoot, outputDir st
 	}
 
 	// Walk and render the template, since input configuration is complete
-	r, err := newRenderer(ctx, config.values, templatePath, libraryPath, outputDir)
+	r, err := newRenderer(ctx, config.values, helpers, templatePath, libraryPath, outputDir)
 	if err != nil {
 		return err
 	}
 	err = r.walk()
 	if err != nil {
 		return err
 	}
-	return r.persistToDisk()
+
+	err = r.persistToDisk()
+	if err != nil {
+		return err
+	}
+	println("✨ Successfully initialized template")
+	return nil
+}
+
+// If the given templateRoot matches
+func prepareBuiltinTemplates(templateRoot string, tempDir string) (string, error) {
+	_, err := fs.Stat(builtinTemplates, path.Join("templates", templateRoot))
+	if err != nil {
+		// The given path doesn't appear to be using out built-in templates
+		return templateRoot, nil
+	}
+
+	// We have a built-in template with the same name as templateRoot!
+	// Now we need to make a fully copy of the builtin templates to a real file system
+	// since template.Parse() doesn't support embed.FS.
+	err = fs.WalkDir(builtinTemplates, "templates", func(path string, entry fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+
+		targetPath := filepath.Join(tempDir, path)
+		if entry.IsDir() {
+			return os.Mkdir(targetPath, 0755)
+		} else {
+			content, err := fs.ReadFile(builtinTemplates, path)
+			if err != nil {
+				return err
+			}
+			return os.WriteFile(targetPath, content, 0644)
+		}
+	})
```

Comment:
Instead of materializing this in a real file system, this should just use the `fs.FS` instance directly.